src/image/TransmissionCurve.cc

```diff
@@ -0,0 +1,827 @@
+//
+class Impl1d {
+public:
+
+    static constexpr bool isSpatiallyConstant = true;
+    static constexpr char const * NAME = "SpatiallyConstantTransmissionCurve";
+
+    // Initialize the GSL interpolator and take ownership of the given arrays.
+    // Array size consistency checks are done by makeSpatiallyConstant,
+    // so we don't need to worry about them here.
+    Impl1d(ndarray::Array<double,1,1> const & throughput,
+           ndarray::Array<double,1,1> const & wavelengths) :
+        _throughput(throughput),
+        _wavelengths(wavelengths),
+        _interp(makeGslPtr(::gsl_interp_alloc(::gsl_interp_linear, _wavelengths.size()),
+                           &::gsl_interp_free))
+    {
+        int status = ::gsl_interp_init(_interp.get(), _wavelengths.getData(), _throughput.getData(),
+                                       _wavelengths.size());
+        LSST_CHECK_GSL(pex::exceptions::LogicError, status);
+    }
+
+    std::pair<double, double> getWavelengthBounds() const {
+        return std::make_pair(_wavelengths.front(), _wavelengths.back());
+    }
+
+    struct Functor {
+
+        Functor(Impl1d const &, geom::Point2D const &) :
+            _accel(makeGslPtr(::gsl_interp_accel_alloc(), &gsl_interp_accel_free))
+        {}
+
+        double operator()(Impl1d const & parent, double wavelength) {
+            double result = 0.0;
+            int status = ::gsl_interp_eval_e(parent._interp.get(), parent._wavelengths.getData(),
+                                             parent._throughput.getData(),
+                                             wavelength, _accel.get(), &result);
+            LSST_CHECK_GSL(pex::exceptions::RuntimeError, status);
+            return result;
+        }
+
+    private:
+        GslPtr<::gsl_interp_accel> _accel;
+    };
+
+    static void setupPersistence(table::Schema & schema, ArrayKeyVector & keys) {
+        keys.push_back(
+            schema.addField<table::Array<double>>(
+                "throughput", "array of known throughput values", "", 0
+            )
+        );
+        keys.push_back(
+            schema.addField<table::Array<double>>(
+                "wavelengths", "array of known wavelength values", "angstrom", 0
+            )
+        );
+    }
+
+    void persist(table::BaseRecord & record, ArrayKeyVector const & keys) const {
+        record.set(keys[0], _throughput);
+        record.set(keys[1], _wavelengths);
+    }
+
+    static Impl1d unpersist(table::BaseRecord & record, ArrayKeyVector const & keys) {
+        return Impl1d(record[keys[0]], record[keys[1]]);
+    }
+
+private:
+    ndarray::Array<double,1,1> _throughput;
+    ndarray::Array<double,1,1> _wavelengths;
+    GslPtr<::gsl_interp> _interp;
+};
+
+//
+// Templated implementation object used by makeRadial.
+//
+class Impl2d {
+public:
+
+    static constexpr bool isSpatiallyConstant = false;
+    static constexpr char const * NAME = "RadialTransmissionCurve";
```

Comment:
Very highly specific?

Reply:
Yes, and I'm afraid that's all these classes are.